### PR TITLE
Update nsis and add file associations

### PR DESCRIPTION
**Pull request by marekr**
this adds file association for .pro files. So they will autoopen kicad ;)

I tried eeschema and pcbnew files as well but it appears they can't handle being passed files like that yet. kicad.exe works fine with pro files.

One thing is we need to copy a icon to the install folder, so not sure how to do that.

### CODE REVIEW

File: nsis/install.nsi [hunk 1]
```diff
@@ -311,6 +314,25 @@ Section $(TITLE_SEC_ENV) SEC07
   SendMessage ${HWND_BROADCAST} ${WM_WININICHANGE} 0 "STR:Environment" /TIMEOUT=5000
 SectionEnd
 
+!macro WRITE_FILE_ASSOC_KEYS EXT EXECUTABLE DESCRIPTION ICON_NAME
+  WriteRegExpandStr ${SOFTWARE_CLASSES_ROOT_KEY} "Software\Classes\.${EXT}\OpenWithProgids\" "${REG_VALUE_NAME}.${EXT}" ""
+
```
Comment (nickoe) [inline nsis/install.nsi]: Why so many empty lines here?

File: nsis/install.nsi [hunk 2]
```diff
@@ -311,6 +314,25 @@ Section $(TITLE_SEC_ENV) SEC07
...
+!macro WRITE_FILE_ASSOC_KEYS EXT EXECUTABLE DESCRIPTION ICON_NAME
+  WriteRegExpandStr ${SOFTWARE_CLASSES_ROOT_KEY} "Software\Classes\.${EXT}\OpenWithProgids\" "${REG_VALUE_NAME}.${EXT}" ""
+  
+  WriteRegExpandStr ${SOFTWARE_CLASSES_ROOT_KEY} "Software\Classes\${REG_VALUE_NAME}.${EXT}" "" "${DESCRIPTION}"
+  
+  WriteRegExpandStr ${SOFTWARE_CLASSES_ROOT_KEY} "Software\Classes\${REG_VALUE_NAME}.${EXT}\" "DefaultIcon" "$INSTDIR\share\kicad\${ICON_NAME}"
+  WriteRegExpandStr ${SOFTWARE_CLASSES_ROOT_KEY} "Software\Classes\${REG_VALUE_NAME}.${EXT}\shell\open\command" "" "$INSTDIR\bin\${EXECUTABLE}"
+!macroend
+
+Section $(TITLE_FILE_ASSOC) SEC08
```
Comment (nickoe) [inline nsis/install.nsi]: Please add MUI_DESCRIPTION_TEXT for SEC08 as is done in MUI_FUNCTION_DESCRIPTION_BEGIN

**Comment by marekr** [comment]
@nickoe updated

File: nsis/install.nsi [hunk 3]
```diff
@@ -451,6 +512,18 @@ Section Uninstall
...
+  ClearErrors
+  ReadRegDWORD $0 ${UNINST_ROOT} "${PRODUCT_UNINST_KEY}" "FileAssocInstalled"
+  IfErrors FinishUninstall 0
+  
+  IntCmp $0 1 0 FinishUninstall FinishUninstall
+  
+  ;delete file associations
+  ${DeleteFileAssociation} "kicad_pcb"
+  ${DeleteFileAssociation} "sch"
+  ${DeleteFileAssociation} "pro"
```
Comment (nickoe) [inline nsis/install.nsi:525]: Shouldn't we delte kicad_wks also?

Comment (marekr) [reply nsis/install.nsi:525]: Woops

**Comment by nickoe** [comment]
Thank you, it looks like it is working. I merged it in:
7339341 Add kicad file association for .pro,.kicad_pcb,.kicad_wks,.sch
c8ff715 Use nsis 3.0.3 instead of the old 3.0 pre-release

**Pull request closed without merging**